src/functions.php

```diff
@@ -601,3 +661,17 @@ function waterfall(array $tasks): PromiseInterface
 
     return $deferred->promise();
 }
+
+/**
+ * @internal
+ */
+function fiberMap(): FiberMap
+{
+    static $wm = null;
```

Comment:
nit: I suppose the name is a leftover from the `WeakMap` implementation?

Reply:
It is, and I intend to introduce that in a follow up PR but left it out for now for the sake it landing this feature.